lib/vanagon/platform/windows.rb

```diff
@@ -289,33 +299,46 @@ def output_dir(target_repo = "")
         File.join("windows", target_repo, @architecture)
       end
 
+      # Generate the underlying directory structure of
+      # any binary files referenced in services. note that
+      # this function does not generate the structure of
+      # the installation directory, only the structure above it.
+      #
+      # @param services, list of all services in a project
+      # @param project, actual project we are creating the directory structure for
+      def generate_service_bin_dirs(services, project)
+        items = []
+        # All service files will need a directory reference
+        services.map do |svc|
+          items.push({
+            :path => strip_and_format_path(svc.service_file, project),
+            :element_to_add => "<Directory Id=\"#{svc.bindir_id}\" />\n"
+          })
+        end
+        return generate_wix_dirs(items)
+      end
+
       # Generate correctly formatted wix elements that match the
-      # structure of the directory input
+      # structure of the itemized input
       #
-      # @param services, Array of components services
-      # and optionally:
+      # @param items
       # @return [string] correctly formatted wix element string
-      def generate_wix_dirs(services)
-        directories = []
-        services.map { |svc| directories.push({ :path => svc.service_file, :bindir_id => svc.bindir_id }) }
+      def generate_wix_dirs(items)
         # root refers to the root of an n-ary tree (which we are about to make)
         root = { :children => [] }
         # iterate over all paths specified and break each one
         # in to its specific directories. This will generate_wix_dirs
         # an n-ary tree structure matching the specs from the input
-        directories.each do |dir|
+        items.each do |item|
           # Always start at the beginning
           curr = root
-          names = strip_path(dir[:path])
-          # The last entry in this list will be the actual file,
-          # which we do not want, we only want it's base path
-          names.pop
+          names = item[:path].split(/\//)
           names.each do |name|
             curr = insert_child(curr, name)
           end
           # at this point, curr will be the top dir, override the id if
           # id exists
-          curr[:bindir_ids].push(dir[:bindir_id])
+          curr[:elements_to_add].push(item[:element_to_add])
         end
         return generate_wix_from_graph(root)
```

Comment:
I'd lose the explicit `return` on a lot of these methods; that's useful when you're exiting early because the work in a method is already done, but if it's the end of the method you should just call/state the value you want to return. In this case, that's `generate_wix_from_graph(root)` without the explicit `return`.